newIDE/app/public/External/Utils/pathEditor.js

```diff
@@ -0,0 +1,132 @@
+const electron = require('electron');
+const fs = require('fs');
+const remote = electron.remote;
+const {
+    dialog
+} = remote;
+
+const headerStyle = {
+    saveFolderLabel: 'height:27px;color:SlateGrey;float: left;margin-left: 2px;margin-top: 10px; font-size:15px;',
+    nameInput: 'font-family:"Courier New";height:27px;width:90px;color:SlateGrey;float:left;margin-left: 2px;padding:4px;margin-top: 4px;font-size:15px;border: 2px solid #e5cd50;border-radius: 3px;  ',
+    fileExistsLabel: 'height:27px;color:blue;float: left;margin-left: 2px;margin-top: 10px; font-size:15px;',
+
+    saveButton: 'height:27px;float:right;margin-left:2px;margin-right:4px;border: 2px solid DeepSkyBlue;border-radius: 1px;margin-top: 5px;background-color:white;',
+    cancelButton: 'height:27px;float:right;margin-right:2px;border: 2px solid DeepSkyBlue;border-radius: 1px;margin-top: 5px;background-color:white;',
+    setFolderButton: 'height:27px;float:right;margin-left:2px;margin-right:4px;border: 2px solid DeepSkyBlue;border-radius: 1px;margin-top: 5px;background-color:white;',
+};
+
+let saveFolderLabel,
+    setFolderButton,
+    nameInput,
+    saveButton,
+    cancelButton,
+    fileExistsLabel = null
+
+export let saveOptions = {}
+let projectBasePath;
+export function loadHeader(parentElement, editorContentDocument, saveToGDFunction, cancelChangesFunction, projectPath, initialResourcePath, extension) {
```

Comment:
if wonder if `loadHeader` should be at all in this file? It is mainly related to jsfx and should I think be either in jsfx-main.js or in a jsfx-header.js file.
This file is called `pathEditor.js` and is put in `Utils/pathEditor.js`, which implies that it can be reused by any external editor, like Piskel - but loadHeader is not generic so we should take it out.

(of course it is calling `updateBasePath`, but that means that it is then not its responsibility to call this, but the responsibility of the caller, i.e: `jsfx-main.js`)

Reply:
I think it would make more sense to move it to jsfx-main and use it as a parameter in loadHeader